### PR TITLE
Cleaned up D2k yaml rules

### PR DESCRIPTION
**Pull request by reaperrr**
Sanitizes defaults and removes some duplication.

Note: #8497 deals with the rest (infantry).

### CODE REVIEW

File: mods/d2k/rules/defaults.yaml [hunk 1]
```diff
@@ -88,68 +90,35 @@
...
-	TargetableUnit:
-		TargetTypes: Ground, C4
-	Passenger:
-		CargoType: Vehicle
-	AttackMove:
-	HiddenUnderFog:
-	GivesExperience:
-	DrawLineToTarget:
-	ActorLostNotification:
 	ProximityCaptor:
```
Comment (penev92) [inline mods/d2k/rules/defaults.yaml]: D2k doesn't need `ProximityCaptor` at all (at the off chance that someone does decide to make a CTF map, they can add the trait in there).
Please remove them all.

File: mods/d2k/rules/defaults.yaml [hunk 2]
```diff
@@ -268,7 +220,8 @@
 		VoiceSet: InfantryVoice
 
 ^Plane:
-	Inherits: ^GainsExperience
+	Inherits@1: ^ExistsInWorld
+	Inherits@2: ^GainsExperience
```
Comment (penev92) [inline mods/d2k/rules/defaults.yaml]: None of the aircraft in D2k gain experience - only Carryalls are buildable, and those are non-combatant.

File: mods/d2k/rules/defaults.yaml [hunk 3]
```diff
@@ -304,7 +248,7 @@
 	Inherits: ^Plane
```
Comment (penev92) [inline mods/d2k/rules/defaults.yaml]: Looks like this is as good a time as any to finally remove `^Helicopter` entirely.

**Comment by reaperrr** [comment]
Rebased & updated. Did a little additional cleanup on aircrafts and aircraft husks, and fixed a few errors caught by Travis/Lint.

**Comment by Mailaender** [comment]
Let's get https://github.com/OpenRA/OpenRA/pull/8497 merged first.

**Comment by reaperrr** [comment]
Rebased.

**Comment by abcdefg30** [comment]
Needs a rebase again.

**Comment by reaperrr** [comment]
Rebased.

**Comment by Mailaender** [comment]
Atreides ornistrike is gone now.

File: mods/d2k/rules/defaults.yaml [hunk 4]
```diff
@@ -1,3 +1,14 @@
+^ExistsInWorld:
+	AppearsOnRadar:
+	UpdatesPlayerStatistics:
+	CombatDebugOverlay:
+	DrawLineToTarget:
```
Comment (GraionDilach) [inline mods/d2k/rules/defaults.yaml]: This was removed from the TS cleanup. Let's aim for consistency.

**Comment by reaperrr** [comment]
@Mailaender: Since building upgrades were introduced, ornistrike requires an upgrade to the hightech factory now (on bleed, too).

**Comment by reaperrr** [comment]
Updated.

**Comment by Mailaender** [comment]
I can't spot regressions. :white_check_mark:

**Comment by penev92** [comment]
Me neither.

**Pull request merged**